pcap.pyx

```diff
@@ -197,8 +202,15 @@ cdef class pcap:
             
         self.__pcap = pcap_open_offline(p, self.__ebuf)
         if not self.__pcap:
-            self.__pcap = pcap_open_live(pcap_ex_name(p), snaplen, promisc,
-                                         timeout_ms, self.__ebuf)
+            self.__pcap = pcap_create(pcap_ex_name(p), self.__ebuf)
+            passing = True
+            pcap_set_snaplen(self.__pcap, snaplen)
+            pcap_set_promisc(self.__pcap, promisc)
+            pcap_set_timeout(self.__pcap, timeout_ms)
+            pcap_set_immediate_mode(self.__pcap, immediate)
+            pcap_set_rfmon(self.__pcap, rfmon)
+            pcap_activate(self.__pcap)
```

Comment:
Good point, will do!

On Aug 14, 2016 17:55, "Guy Harris" <notifications@github.com> wrote:

> In pcap.pyx
> <https://github.com/pynetwork/pypcap/pull/32#discussion_r74712447>:
>
> > @@ -197,8 +202,15 @@ cdef class pcap:
> >
> >          self.__pcap = pcap_open_offline(p, self.__ebuf)
> >          if not self.__pcap:
> > -            self.__pcap = pcap_open_live(pcap_ex_name(p), snaplen, promisc,
> > -                                         timeout_ms, self.__ebuf)
> > +            self.__pcap = pcap_create(pcap_ex_name(p), self.__ebuf)
> > +            passing = True
> > +            pcap_set_snaplen(self.__pcap, snaplen)
> > +            pcap_set_promisc(self.__pcap, promisc)
> > +            pcap_set_timeout(self.__pcap, timeout_ms)
> > +            pcap_set_immediate_mode(self.__pcap, immediate)
> > +            pcap_set_rfmon(self.__pcap, rfmon)
> > +            pcap_activate(self.__pcap)
>
> You should check for errors here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pynetwork/pypcap/pull/32/files/0a0962d0a7ff0a71772256c1e44770f7d7eb2c07#r74712447>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAOnrWjL_yQLZBx9j9bV-HilY0kOuZujks5qf7jrgaJpZM4JkCjY>
> .
>